### DISCUSSION TITLE
How can I add a Letsencrypt certificate to make https connections secure?

### DISCUSSION THREAD

**Discussion by GustavBrock** [Azure/data-api-builder#1888] [Q&A] [answered]
I launch dab.exe on a Windows Server 2022 exposed publicly via one of our domains.
How can I install a certificate from Letsencypt to make https-connections to my API secure?

**Comment by seantleonard** [comment]
I suggest taking a look through ASP.NET's hosting documentation to determine the hosting mechanism the best suits your needs. There are also instructions on how to set up your environment for each of those scenarios:
https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/?view=aspnetcore-6.0

**Reply by GustavBrock** [reply]
Thanks, a lot of options.
We have IIS running. Could I run dab.exe "as is" behind IIS set up as a reverse proxy? That, I guess, would mean that the certificate should be installed with IIS which seems well documented.

**Reply by GustavBrock** [reply]
I have tried running DAB using IIS as a reverse proxy. It works with http, but I can see no method to have a certificate attached, thus no https.
So, would there be a simpler setup, forgetting Letsencrypt for a moment? 
Running DAB standalone, how can I assign a certificate to DAB to be seen by clients connecting via https? No authentication is needed.

**Reply by pholcroft** [reply, marked as answer]
I believe the focus with DAB is on the API functionality itself, not the infrastructure such as certificates etc.
You should be able to achieve what you want by using Acme to handle letsencrypt with IIS.
Alternatively, I recommend an easier route of using a [cloudflare tunnel](https://www.cloudflare.com/en-gb/products/tunnel/) and their [managed SSL service](https://www.cloudflare.com/en-gb/application-services/products/ssl/). All of this can be achieved on their free tier unless you need specific functionality. All you need is your own domain.

**Reply by GustavBrock** [reply]
@pholcroft : I gave up on the IIS route. "Should be able ...", perhaps, but whatever I tried, something final missed.
However, Cloudflare delivers - and for free for a small setup. 

I moved a domanin of ours (you _will_ need a spare or a new domain), sat up a tunnel (which sets the subdomain to call), installed the Cloudflared service, and - _boo_ ... 

The final steps to carry out for having **dab** to respond:

- Tunnels
- Your Tunnel name: Configure (the three dots at right)
- Public Hostname
- Menu: Edit
- Additional application settings
- TLS
- No TLS Verify: Enabled
- Save hostname

Success. 
Thank you so much!

**Comment by julianadormon** [comment]
I also had no luck with IIS and reverse proxy. Kept getting a 502 error. Cloudflare tunnel worked like a charm.
